lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -3042,21 +3042,31 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       const bool wasAlreadyInvalid = ED->isInvalid();
       ED->setInvalid();
       if (!extType->hasError() && extType->getAnyNominal()) {
+        auto canExtType = extType->getCanonicalType();
+        if (auto existential = canExtType->getAs<ExistentialType>()) {
```

Comment:
You can call getAs() on a non-canonical type, it will automatically desugar it, so you don’t need to move the canonical type computation from below

Reply:
Using `auto existential = extType->getAs<ExistentialType>()` instead still gives a difference. The comment for the fixit becomes `"did you mean to extend 'A4' (aka 'P4') instead?"`, while it will still replace the type with just `"P4"`. Wouldn't it be better if the comment just says `"did you mean to extend 'P4' instead?"`?

Reply:
I personally think it's better for the error message to mention `'A4' (aka 'P4')` because `A4` is what's written in the extension, so it helps the programmer make the connection between `A4` and `P4`. However, I don't feel too strongly about this and I'm fine with merging this change as-is.